da/types/sidecar.go

```diff
@@ -159,25 +159,17 @@ func (b *BlobSidecar) MarshalSSZ() ([]byte, error) {
 	return buf, ssz.EncodeToBytes(buf, b)
 }
 
-// empty creates an empty BlobSidecar object.
-func (*BlobSidecar) empty() *BlobSidecar {
-	return &BlobSidecar{
-		SignedBeaconBlockHeader: &ctypes.SignedBeaconBlockHeader{},
-	}
-}
-
-// NewFromSSZ creates a new BlobSidecar object from SSZ format.
-func (b *BlobSidecar) NewFromSSZ(buf []byte) (*BlobSidecar, error) {
-	b = b.empty()
-	if err := ssz.DecodeFromBytes(buf, b); err != nil {
-		return nil, err
+func (b *BlobSidecar) EnsureSyntaxFromSSZ() error {
+	// Ensure SignedBeaconBlockHeader is not nil
+	if b.SignedBeaconBlockHeader == nil {
+		b.SignedBeaconBlockHeader = &ctypes.SignedBeaconBlockHeader{}
```

Comment:
not sure this is needed anymore, but keeping it for good measure.
This is the reason these methods are called `Ensure` instead of just `Check`. I intend to signal that they can modify the pointer in place, not just read from it and to checks